src/entities/key-value-pair.ts

```diff
@@ -0,0 +1,4 @@
+export interface KeyValuePair<K, V> {
```

Comment:
May not be needed, check if you can use [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) instead